server/src/test/java/org/opensearch/indices/recovery/RecoverySourceHandlerTests.java

```diff
@@ -215,7 +215,7 @@ public void writeFileChunk(
                 });
             }
         };
-        RecoverySourceHandler handler = new RecoverySourceHandler(
+        RecoverySourceHandler handler = new DefaultRecoverySourceHandler(
```

Comment:
These tests essentially become tests for `DefaultRecoverySourceHandler`. Don't we need tests for `RemoteStoreReplicaRecoverySourceHandler` as well?

Reply:
The `RecoverySourceHandlerTests` now covers both the RecoverySourceHandler and DefaultRecoverySourceHandler. The tests for RemoteStoreReplicaRecoverySourceHandler has were written in an older PR and is present in `ReplicaRecoveryWithRemoteTranslogOnPrimaryTests.java`.  Let me decompose it and create another test class dedicated to the new Handler.